### PR TITLE
Safe creation

### PR DESCRIPTION
**Pull request by germartinez**
Allows to deploy Safe accounts using the `SafeFactory` class.

### CODE REVIEW

File: packages/safe-core-sdk/src/configuration/contracts.ts [hunk 1]
```diff
@@ -11,30 +15,44 @@ export interface ContractNetworksConfig {
...
   1: {
-    multiSendAddress: '0x8D29bE29923b68abfDD21e541b9374737B49cdAD'
+    multiSendAddress: '0x8D29bE29923b68abfDD21e541b9374737B49cdAD',
+    safeMasterCopyAddress: '0x6851D6fDFAfD08c0295C392436245E5bc78B0185',
+    safeProxyFactoryAddress: '0x76E2cFc1F5Fa8F6a5b3fC4c8F4788F0116861F9B'
   },
   // rinkeby
   4: {
-    multiSendAddress: '0x8D29bE29923b68abfDD21e541b9374737B49cdAD'
+    multiSendAddress: '0x8D29bE29923b68abfDD21e541b9374737B49cdAD',
```
Comment (mikheevm) [inline packages/safe-core-sdk/src/configuration/contracts.ts:24]: can you get this info from safe-deployments package?

File: packages/safe-core-sdk/src/safeFactory/utils.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { SafeAccountConfig } from './'
+
+export const validateSafeAccountConfig = ({ owners, threshold }: SafeAccountConfig): void => {
+  if (owners.length <= 0) throw new Error('Owner list must have at least one owner')
+  if (threshold <= 0) throw new Error('Threshold must be greater than or equal to 0')
```
Comment (mikheevm) [inline packages/safe-core-sdk/src/safeFactory/utils.ts:5]: https://github.com/gnosis/safe-contracts/blob/186a21a74b327f17fc41217a927dea7064f74604/contracts/base/OwnerManager.sol#L29
```suggestion
  if (threshold <= 0) throw new Error('Threshold must be greater than or equal to 1')
```

File: packages/safe-core-sdk/src/contracts/GnosisSafeProxyFactory/GnosisSafeProxyFactoryWeb3Contract.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    } else {
+      txResponse = this.contract.methods
+        .createProxy(safeMasterCopyAddress, initializer)
+        .send(options)
+    }
+    const txResult: TransactionReceipt = await new Promise((resolve, reject) =>
+      txResponse.once('receipt', (receipt: TransactionReceipt) => resolve(receipt)).catch(reject)
+    )
+    const proxyAddress = '0x' + txResult.events?.['0'].raw?.data.substr(-40)
+    if (!proxyAddress) {
```
Comment (mikheevm) [inline packages/safe-core-sdk/src/contracts/GnosisSafeProxyFactory/GnosisSafeProxyFactoryWeb3Contract.ts:32]: "0x" + undefined will never be falsy so this if block will never work, you need to check the contents after "0x" or prepend "0x" inside the if statement

File: packages/safe-core-sdk/src/contracts/GnosisSafeProxyFactory/GnosisSafeProxyFactoryEthersV5Contract.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        safeMasterCopyAddress,
+        initializer,
+        saltNonce,
+        options
+      )
+    } else {
+      txResponse = await this.contract.createProxy(safeMasterCopyAddress, initializer, options)
+    }
+    const txReceipt = await txResponse.wait()
+    const proxyCreationEvent = txReceipt.events?.find(({ event }: any) => event === 'ProxyCreation')
```
Comment (mikheevm) [inline packages/safe-core-sdk/src/contracts/GnosisSafeProxyFactory/GnosisSafeProxyFactoryEthersV5Contract.ts:30]: the type could be more specific there I think

**Review by mikheevm — APPROVED** [review]
💪

File: packages/safe-core-sdk/src/configuration/contracts.ts (hunk 1 above)
Comment (germartinez) [reply packages/safe-core-sdk/src/configuration/contracts.ts:24]: Here :) https://github.com/gnosis/safe-core-sdk/issues/25

**Pull request merged**